### PR TITLE
Add container optimizations

### PR DESCRIPTION
**Pull request by simonkurtz-MSFT**
Slotting this as module 12.

### CODE REVIEW

**Comment by simonkurtz-MSFT** [comment]
@richlander & @baronfel, would very much appreciate if you could take a look at above PR Preview URL for Module 99, please.

File: docs/aca/99-optimize-containers/Backend.Api.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM mcr.microsoft.com/dotnet/aspnet:8.0 AS base
+WORKDIR /app
+EXPOSE 5000
+
+ENV ASPNETCORE_URLS=http://+:5000
```
Comment (richlander) [inline docs/aca/99-optimize-containers/Backend.Api.Dockerfile:5]: That works. .NET 8 has a simpler ENV -- `ASPNETCORE_HTTP_PORTS` -- for doing the same thing.

https://github.com/dotnet/dotnet-docker/blob/dbad8de557e0f943fe5103f9c165ac3300832d97/src/runtime-deps/8.0/jammy-chiseled/amd64/Dockerfile#L47

Comment (simonkurtz-MSFT) [reply docs/aca/99-optimize-containers/Backend.Api.Dockerfile:5]: Wondering whether all target ports should switch to 8080 then with .NET 8.

Comment (simonkurtz-MSFT) [reply docs/aca/99-optimize-containers/Backend.Api.Dockerfile]: @richlander, I opened #148 to address the port separately.

**Review by waelkdouh — APPROVED** [review]
LGTM

**Pull request merged**